src/common/pages/sign-up.tsx

```diff
@@ -49,12 +48,19 @@ class SignUpPage extends Component<PageProps, State> {
   };
 
   componentDidMount() {
-    const { location } = this.props;
+    const { location, history } = this.props;
     const qs = queryString.parse(location.search);
     if (qs.referral) {
       const referral = qs.referral as string;
       this.setState({ referral, lockReferral: true });
     }
+    if (ls.get("referral")) {
```

Comment:
if url param has referral we should make that priority than ls.

Reply:
done @feruzm kindly check